spyderlib/otherplugins.py

```diff
@@ -8,52 +8,115 @@
 Spyder third-party plugins configuration management
 """
 
-import os
-import os.path as osp
+import importlib
 import sys
 import traceback
 
 # Local imports
-from spyderlib.utils import programs
-
-
-# Calculate path to `spyderplugins` package, where Spyder looks for all 3rd
-# party plugin modules
-PLUGIN_PATH = None
-if programs.is_module_installed("spyderplugins"):
-    import spyderplugins
-    PLUGIN_PATH = osp.abspath(spyderplugins.__path__[0])
-    if not osp.isdir(PLUGIN_PATH):
-        # py2exe/cx_Freeze distribution: ignoring extra plugins
-        PLUGIN_PATH = None
-
-
-def get_spyderplugins(prefix, extension):
-    """Scan directory of `spyderplugins` package and
-    return the list of module names matching *prefix* and *extension*"""
-    plist = []
-    if PLUGIN_PATH is not None:
-        for name in os.listdir(PLUGIN_PATH):
-            modname, ext = osp.splitext(name)
-            if prefix is not None and not name.startswith(prefix):
-                continue
-            if extension is not None and ext != extension:
-                continue
-            plist.append(modname)
-    return plist
-
-
-def get_spyderplugins_mods(prefix, extension):
-    """Import modules that match *prefix* and *extension* from
-    `spyderplugins` package and return the list"""
-    modlist = []
-    for modname in get_spyderplugins(prefix, extension):
-        name = 'spyderplugins.%s' % modname
-        try:
-            __import__(name)
-            modlist.append(sys.modules[name])
-        except Exception:
-            sys.stderr.write(
-                "ERROR: 3rd party plugin import failed for `%s`\n" % modname)
-            traceback.print_exc(file=sys.stderr)
+from spyderlib.config.base import get_conf_path
+from spyderlib.py3compat import PY2, PY3, PY33
+from spyderlib.utils.external.path import Path
+
+if PY2:
+    import imp
+
+INIT_PY = """# -*- coding: utf-8 -*-
+
+# Declare as a namespace package
+__import__('pkg_resources').declare_namespace(__name__)
+"""
+
+
+def _get_spyderplugins(plugin_path, base_namespace, plugins_namespace,
+                       modnames, modlist):
+    """Scan the directory `plugin_path` for plugins_namespace package and
+    loads its submodules."""
+    namespace_path = Path(plugin_path) / base_namespace / plugins_namespace
+
+    if not namespace_path.exists():
+        return
+
+    for dirname in namespace_path.dirs():
+        if dirname.name == "__pycache__":
+            continue
+        _import_plugin(dirname.name, base_namespace, plugins_namespace,
+                       namespace_path, modnames, modlist)
+
+    for name in namespace_path.files(pattern="*.py"):
+        if name.name == "__init__.py":
+            continue
+        _import_plugin(name.namebase, base_namespace, plugins_namespace,
+                       namespace_path, modnames, modlist)
+
+
+class _ModuleMock():
+    """ """

```

Comment:
Oops I have to add the docstring....

It was bit tricky and the only way I could get around this... so

When we are loading a plugin it will try to load the translations, but since the get_translations methods depends on a module already loaded in `sys.modules` it was failing, so basically this class is just a mock I use to store a class variable `LOCALEPATH` that get transtalations will use to correctlyfin the files.. once the module loads, I restore the actual module object into `sys.modules`